spec/scenarios/meta_data_by_context_spec.rb

```diff
@@ -0,0 +1,130 @@
+describe 'querying Metadata by Context', type: :request do
+  %w{0 1 2}.each do |n|
+    context_key = "context_key_#{n}"
+    let(context_key) { FactoryGirl.create(:context_key) }
+    let("context_#{n}") { send(context_key).context }
```

Comment:
Why do you use `send` at all?

Reply:
To call method defined in line 4 by it's name defined as `context_key` string in line 3.
What would be better way to do that?

Reply:
I'd prefer to replace the loop with explicitly defined `let`s, I think it's easier to read. My previous notes were because of that.

Reply:
```
    let(:context_key_0) { FactoryGirl.create(:context_key) }
    let(:context_0) { context_key_0.context }
    let(:meta_datum_0) { FactoryGirl.create(:meta_datum_title, meta_key: context_key_0.meta_key) }
    let(:context_key_1) { FactoryGirl.create(:context_key) }
    let(:context_1) { context_key_1.context }
    let(:meta_datum_1) { FactoryGirl.create(:meta_datum_title, meta_key: context_key_1.meta_key) }
    let(:context_key_2) { FactoryGirl.create(:context_key) }
    let(:context_2) { context_key_2.context }
    let(:meta_datum_2) { FactoryGirl.create(:meta_datum_title, meta_key: context_key_2.meta_key) }
```
Like that? :/ Not too dry... That's why I came up with this loop. But on the other hand you're right that it may be easier to read

Reply:
IMHO `let(:meta_datum_#)`s should be omitted as they're used only in method `set_meta_datum_and_contexts`. The method should take care of creating `meta_datum`s instead. That way you'll use ~33% of `let` calls ;)

Reply:
It would be good but this method is also called in expectation in line 68 `expect(kontext_meta_data_node[:metaDatum].map(&:values).flatten).
        to include(send("meta_datum_#{i}").id)`

Reply:
Mhmm, ok I give up...

Reply:
I changed it. It's few lines more but it's much more readable now :)